### PR TITLE
[10.2.X] update lhapdf sets to 6.4.0e

### PR DESCRIPTION
**Pull request by aandvalenzuela**
backport of https://github.com/cms-sw/cmsdist/pull/7776

### CODE REVIEW

**Comment by aandvalenzuela** [comment]
please test

File: lhapdf.spec [hunk 1]
```diff
@@ -1,27 +1,25 @@
...
-Source6: https://www.hepforge.org/archive/lhapdf/pdfsets/6.1/MMHT2014nlo68cl.tar.gz
+Source2: http://www.hepforge.org/archive/lhapdf/pdfsets/6.1/MSTW2008nlo68cl.tar.gz
 
-Source7: lhapdf_pdfsetsindex
+Source3: lhapdf_pdfsetsindex
 
-Requires: python
-BuildRequires: py2-cython
+Requires: python3
+BuildRequires: py3-cython
```
Comment (smuzaffar) [inline lhapdf.spec]: @aandvalenzuela , for 10.2.X, we should still use `python and py2-cython` dep. So please revert this change above

File: lhapdf.spec [hunk 2]
```diff
@@ -1,27 +1,25 @@
...
 
 %define keep_archives true
 
 %prep
 %setup -q -n LHAPDF-%{realversion}
 
-./configure --prefix=%{i} 
+PYTHON=$(which python3) \
```
Comment (smuzaffar) [inline lhapdf.spec]: @aandvalenzuela , also remove this line or use `PYTHON=$(which python)` . For 10.2.X , we still have python ( i.e. python2) as default

**Comment by aandvalenzuela** [comment]
please test

**Comment by aandvalenzuela** [comment]
@smuzaffar I think the tests were successful, but not correctly reported ([relvals](https://cmssdt.cern.ch/jenkins/job/ib-run-pr-wait-deployment/32674/), [addon](https://cmssdt.cern.ch/jenkins/job/ib-run-pr-wait-deployment/32671/) and [unittest](https://cmssdt.cern.ch/jenkins/job/ib-run-pr-tests/25992/)).

**Comment by smuzaffar** [comment]
@aandvalenzuela , [addon](https://cmssdt.cern.ch/jenkins/job/ib-run-pr-addon/) and [relvals](https://cmssdt.cern.ch/jenkins/job/ib-run-pr-relvals/) jobs are pending in jenkins for this PR. lets wait for those to finish first. bot should report the test results once all the pending tests are done

**Comment by smuzaffar** [comment]
+externals
@qliphy  @perrotta  this is backport of lhapdf 6.4.0e for 10.2.X IB/Release. Feel free to include this in next 10.2.X release.

**Comment by smuzaffar** [comment]
by the way , the unit tests failures are not related to this PR. These tests are already failing for 10.2.X IBs  ( https://cmssdt.cern.ch/SDT/cgi-bin/showBuildLogs.py/slc7_amd64_gcc700/www/sun/10.2-sun-00/CMSSW_10_2_X_2022-07-03-0000?utests )

**Comment by qliphy** [comment]
merge
